### ISSUE TITLE
Global alignment

### ISSUE THREAD

**Issue by haesoochung** [guanyingc/DeepHDRVideo#13]
Hello. According to your paper, global alignment is performed using a similarity transformation in the preprocessing stage. I wonder if this global alignment process is performed during training phase in the code because I couldn't find it. Thank you!

**Comment by guanyingc** [comment]
Hi, during training with synthetic data, we don't use global alignment to increase difficulty of training data.

**Comment by haesoochung** [comment]
Thank you! May I ask you one more thing? Did you use all the options as default setting?

**Comment by guanyingc** [comment]
Yes, the provided code is the setting for training.

**Comment by haesoochung** [comment]
Thank you :)

**Comment by haesoochung** [comment]
Hi. Following the implementation details mentioned in your paper, I trained the model using the uploaded code without any modification, but I couldn’t get the result in your paper... (Especially 2 exposures model) 😭 Should I change batch size or # of training epoch according to 2E/3E model? Thank you in advance.

**Comment by guanyingc** [comment]
Hi, what result did you get? How far is it from the result reported the paper.

**Comment by haesoochung** [comment]
For example, on the synthetic test dataset, I got PSNRs as follows:
2 exposures: 28.35 (vs. 40.34 in the paper)
3 exposures: 33.68 (vs. 37.04 in the paper)
I used the test code following the instructions.
I trained the CoarseNet with 10 epochs using a batch size of 16, trained the RefineNet with 15 epochs using
a batch size of 8, and end-to-end finetuned the whole network for 2 epochs using a  batch size of 8...

**Comment by guanyingc** [comment]
Oh, there might be some problems in the training. Could you check the PSNR results of stage1 and stage2? Not sure which stage goes wrong.

**Comment by haesoochung** [comment]
After stage1:
---------- TRAIN Epoch 10 Summary -----------
	 [LOSS] hdr_loss: 0.00270| 
	 [PSNR] hdr_psnr: 45.01| high_psnr: 45.01| 
	 [LSNR] hdr_lsnr: 35.64| high_lsnr: 35.64| 
	 [SSIM] hdr_ssim: 0.9907| high_ssim: 0.9907| 
	 [RANGE] f_min_range: -20.88| f_mean_range: 1.93| f_max_range: 24.87| 
	 [HIGH] high_psnr: 45.01| high_lsnr: 35.64| high_ssim: 0.99| 

---------- VAL Epoch 10 Summary -----------
	 [PSNR] hdr_psnr: 37.65| low_psnr: 30.86| high_psnr: 44.55| 
	 [LSNR] hdr_lsnr: 49.45| low_lsnr: 63.03| high_lsnr: 35.75| 
	 [SSIM] hdr_ssim: 0.8010| low_ssim: 0.6150| high_ssim: 0.9897| 
	 [RANGE] f_min_range: -24.46| f_mean_range: 2.22| f_max_range: 23.58| 
	 [LOW] low_psnr: 30.86| low_lsnr: 63.03| low_ssim: 0.62| 
	 [HIGH] high_psnr: 44.55| high_lsnr: 35.75| high_ssim: 0.99| 

After stage2:
	---------- TRAIN Epoch 15 Summary -----------
	 [LOSS] hdr_loss: 0.00408| mhdr_loss: 0.00878| perc_loss: 0.31879| 
	 [ERR] hdr0_err: 0.00060| hdr1_err: 0.01686| hdr2_err: 0.00060| 
	 [PSNR] hdr_psnr: 44.78| low_psnr: 44.78| hdr2_psnr: 38.40| low2_psnr: 38.40| 
	 [LSNR] hdr_lsnr: 66.99| low_lsnr: 66.99| hdr2_lsnr: 67.44| low2_lsnr: 67.44| 
	 [SSIM] hdr_ssim: 0.9895| low_ssim: 0.9895| hdr2_ssim: 0.8202| low2_ssim: 0.8202| 
	 [RANGE] f_min_range: -19.92| f_mean_range: 1.70| f_max_range: 21.03| 
	 [LOW] low_psnr: 44.78| low_lsnr: 66.99| low_ssim: 0.99| low2_psnr: 38.40| low2_lsnr: 67.44| low2_ssim: 0.82| 

---------- VAL Epoch 15 Summary -----------
	 [PSNR] hdr_psnr: 37.50| low_psnr: 45.17| high_psnr: 29.91| hdr2_psnr: 41.74| low2_psnr: 37.62| high2_psnr: 45.76| 
	 [LSNR] hdr_lsnr: 43.63| low_lsnr: 66.46| high_lsnr: 21.01| hdr2_lsnr: 52.35| low2_lsnr: 67.49| high2_lsnr: 37.33| 
	 [SSIM] hdr_ssim: 0.9648| low_ssim: 0.9892| high_ssim: 0.9406| hdr2_ssim: 0.8630| low2_ssim: 0.7335| high2_ssim: 0.9905| 
	 [RANGE] f_min_range: -20.89| f_mean_range: 1.49| f_max_range: 18.86| 
	 [LOW] low_psnr: 45.17| low_lsnr: 66.46| low_ssim: 0.99| low2_psnr: 37.62| low2_lsnr: 67.49| low2_ssim: 0.73| 
	 [HIGH] high_psnr: 29.91| high_lsnr: 21.01| high_ssim: 0.94| high2_psnr: 45.76| high2_lsnr: 37.33| high2_ssim: 0.99| 

After fine-tuning:
---------- TRAIN Epoch 2 Summary -----------
	 [LOSS] hdr_loss: 0.00394| mhdr_loss: 0.00850| perc_loss: 0.30743| 
	 [ERR] hdr0_err: 0.00075| hdr1_err: 0.01613| hdr2_err: 0.00075| 
	 [PSNR] hdr_psnr: 45.22| low_psnr: 45.22| hdr2_psnr: 35.99| low2_psnr: 35.99| 
	 [LSNR] hdr_lsnr: 66.25| low_lsnr: 66.25| hdr2_lsnr: 64.62| low2_lsnr: 64.62| 
	 [SSIM] hdr_ssim: 0.9899| low_ssim: 0.9899| hdr2_ssim: 0.8498| low2_ssim: 0.8498| 
	 [RANGE] f_min_range: -24.07| f_mean_range: 1.92| f_max_range: 23.04| 
	 [LOW] low_psnr: 45.22| low_lsnr: 66.25| low_ssim: 0.99| low2_psnr: 35.99| low2_lsnr: 64.62| low2_ssim: 0.85| 

---------- VAL Epoch 2 Summary -----------
	 [PSNR] hdr_psnr: 37.64| low_psnr: 45.07| high_psnr: 30.26| hdr2_psnr: 39.08| low2_psnr: 33.69| high2_psnr: 44.36| 
	 [LSNR] hdr_lsnr: 43.79| low_lsnr: 66.09| high_lsnr: 21.65| hdr2_lsnr: 49.36| low2_lsnr: 63.36| high2_lsnr: 35.46| 
	 [SSIM] hdr_ssim: 0.9671| low_ssim: 0.9891| high_ssim: 0.9451| hdr2_ssim: 0.8649| low2_ssim: 0.7396| high2_ssim: 0.9880| 
	 [RANGE] f_min_range: -17.32| f_mean_range: 1.49| f_max_range: 20.51| 
	 [LOW] low_psnr: 45.07| low_lsnr: 66.09| low_ssim: 0.99| low2_psnr: 33.69| low2_lsnr: 63.36| low2_ssim: 0.74| 
	 [HIGH] high_psnr: 30.26| high_lsnr: 21.65| high_ssim: 0.95| high2_psnr: 44.36| high2_lsnr: 35.46| high2_ssim: 0.99| 

Is something wrong?..

**Comment by guanyingc** [comment]
Thanks for the information. The provided information are for training and evaluation. Could you show the results for the testing.

**Comment by haesoochung** [comment]
Testing on the synthetic test dataset, 
after stage 1:  32.43 / stage 2: 28.38
It seems that something went wrong during training stage 2..

**Comment by guanyingc** [comment]
If the result in the paper is 40.34, then both stage 1 and stage2 might have some problems. The result of stage1 should be be something like 38. 

I don't know what is the problem. Have you strictly follow the training guidance? Have you used both the vimeo and hdr dataset for training?

**Comment by haesoochung** [comment]
Oh.. Yes, I used both datasets.
I was checking the training log of stage 1, and I just find out that learning rate starts from 0.0001 and changes to 2.5e-05 at epoch 4, 5e-05 at epoch 5, 1.25e-05 at epoch 10.. Is it right?

**Comment by guanyingc** [comment]
I think this learning rate scedule is ok. In https://github.com/guanyingc/DeepHDRVideo/issues/18, someone has reported the results trained using only the HDR dataset, but still achieves better results than yours.

**Comment by haesoochung** [comment]
I'm trying to found out the problem. Thanks a lot!!
